library/libraries.js

```diff
@@ -1753,5 +1753,37 @@ var d41d8cd98f00b204e9800998ecf8427e_LibraryDetectorTests = {
 
             return false;
         }
-    }
+    },
+    'Joomla': {
+        id: 'joomla',
+        icon: 'joomla',
+        url: 'https://www.joomla.org/',
+        npm: null,
+        test: async function (win) { 
+             const generatorMeta = !!document.querySelector('meta[name="generator"][content*="Joomla"]');
+             const joomlaComponent = !!document.querySelector('[href*="index.php?option=com_"]');
+
+             try {
+                const response = await fetch('/language/en-GB/en-GB.xml');
```

Comment:
Technical this is the version of the language en-GB. Not from joomla core.

I would suggest to first check the manifest from here: 

/administrator/mainifests/files/joomla.xml

When that fails (because there is a htaccess protection for example) you can still fallback to the language xml.

Reply:
Ooh good point, I did notice that `joomla.xml` can be blocked so I just used `en-GB.xml`. Having that as an initial check before falling back to the language xml makes more sense though!

@jeckodevelopment would you be okay with including this?

Reply:
I generally use the en-GB.xml language file to detect the version, as it's updated at every release.
Many automated tools use this method

Reply:
Well right now that works good but when we switch back to the language versioning (3.9.18.1 would than be the correct version) this might be a problem. (right now there is no intention to do so)

As this is technical speaking the version of the language and not the cms.

Thats the reason i would suggest when this is a test for Lighthouse they should try to detect the version from the right spot in my opinion.

You can still use the language xml as a fallback.

Also please keep in mind that the language xml file has been renamend in joomla 4 so this check would not detect the version starting from Joomla 4.0

Reply:
> Ooh good point, I did notice that `joomla.xml` can be blocked so I just used `en-GB.xml`. Having that as an initial check before falling back to the language xml makes more sense though!
> 
> @jeckodevelopment would you be okay with including this?

Technical speaking both files can be blocked by a simple htacces rule. Disallowing access to xml files.

The only way to check Joomla without that it can be blocked is to check known media files to be existent. So you get an idea that it might be joomla 2.5.x 3.x or 4.x  but not the exact version. There fore more advaned checks needs to be done.

Reply:
Can you please help writing such checks?